mobile/lib/modules/backup/background_service/background.service.dart

```diff
@@ -344,35 +350,43 @@ class BackgroundService {
     AppSettingsService settingService = AppSettingsService();
     BackupService backupService = BackupService(apiService, db, settingService);
     AppSettingsService settingsService = AppSettingsService();
-
-    final selectedAlbums = backupService.selectedAlbumsQuery().findAllSync();
-    final excludedAlbums = backupService.excludedAlbumsQuery().findAllSync();
-    if (selectedAlbums.isEmpty) {
-      return true;
-    }
+    HashService hashService = HashService(db, this);
+    SyncService syncService = SyncService(db);
+    BackupAlbumService backupAlbumService = BackupAlbumService(db);
+    LocalAlbumService localAlbumService =
+        LocalAlbumService(db, hashService, syncService, backupAlbumService);
 
     await PhotoManager.setIgnorePermissionCheck(true);
 
     do {
+      await localAlbumService.refreshDeviceAlbums();
```

Comment:
We need a new lock:

This function must not run concurrently with any sync operation in the foreground app.

So...we probably need to add the backup lock to the sync service (so it becomes a two-step lock, first either FG or BG gains lock, then individual tasks can run sequentially)